data/meson.build

```diff
@@ -1,6 +1,18 @@
 config = configuration_data()
 config.set('app_id', app_id)
 
+pkgdatadir = join_paths(get_option('prefix'), get_option('datadir'), meson.project_name())
```

Comment:
Is there are reason other prefix is added here? Other paths don't use it (I think it's added automatically but I might be wrong). Having that I mind, if there is no need for prefix here, the `join_paths` method could just be put in `compile_resources` method like in other places where it's used.

Reply:
It's used in the install_dir kwarg of a target, but install_dir will automatically use paths relative to prefix when given a relative path. So you do not need to join it to the prefix manually (and in fact it's better not to, because Meson can trace the relationship better in its internal model).

You would usually want to add the prefix if you're inserting the value into a configuration_data though.